### PR TITLE
example(GraphQuestions): add graph_questions_testing dataset

### PR DESCRIPTION
**Pull request by anda-ren**
## Description
1. add graph_questions_testing dataset
2.  add dataset util function `from_dict`

## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/api/_impl/dataset/model.py [hunk 1]
```diff
@@ -998,3 +998,51 @@ def dataset(
         )
 
         return ds
+
+    @classmethod
+    def from_dict(
+        cls, name: str = "", data: dict = {}, field_selector: str = ""
```
Comment (tianweidut) [inline client/starwhale/api/_impl/dataset/model.py:1004]:
```
name: (str, required) The dataset name you would like to use.
```
so we make name argument required?  `name: str=""` -> `name: str` ?

Comment (tianweidut) [reply client/starwhale/api/_impl/dataset/model.py:1004]: `data: dict = {}` --> `data: dict`  or `data: t.Optional[t.Dict] = None`?

File: client/starwhale/api/_impl/dataset/model.py [hunk 2]
```diff
@@ -998,3 +998,51 @@ def dataset(
...
+        """
+
+        data_items = data
+        if field_selector:
+            # Split field selector by dots
+            fields = field_selector.split(".")
+            # Iterate over selected fields
+            for field in fields:
+                if field in data_items:
+                    data_items = data_items[field]
```
Comment (tianweidut) [inline client/starwhale/api/_impl/dataset/model.py:1034]: data = {"a": 1, "b": {"c": 1}}

can we select `data["b"]["c"]` ?

File: client/starwhale/api/_impl/dataset/model.py [hunk 3]
```diff
@@ -998,3 +998,51 @@ def dataset(
...
+            ```python
+            from starwhale import Dataset
+            myds = Dataset.from_dict("translation", [{"en":"hello","zh-cn":"你好"},{"en":"how are you","zh-cn":"最近怎么样"}])
+            print(myds[0].features.en)
+            ```
+            ```python
+            from starwhale import Dataset
+            myds = Dataset.from_dict("translation", {"content":{"child_content":[{"en":"hello","zh-cn":"你好"},{"en":"how are you","zh-cn":"最近怎么样"}]}},"content.child_content")
+            print(myds[0].features["zh-cn"])
+            ```
```
Comment (tianweidut) [inline client/starwhale/api/_impl/dataset/model.py:1024]: add a example for `field_selector`?

Comment (anda-ren) [reply client/starwhale/api/_impl/dataset/model.py:1023]: https://github.com/star-whale/starwhale/blob/ff6a81c8ab88bc8ebb0c79e49841dbd9844ef563/client/starwhale/api/_impl/dataset/model.py#L1021
"content.child_content" is an example

File: client/starwhale/api/_impl/dataset/model.py (hunk 1 above)
Comment (anda-ren) [reply client/starwhale/api/_impl/dataset/model.py]: done

**Review by tianweidut — APPROVED** [review]
LGTM

**Pull request merged**